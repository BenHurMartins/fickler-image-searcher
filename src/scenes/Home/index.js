import React from 'react';
import {
  ViewContainer,
  SearchBarComponent,
  FlatListComponent,
  LoadingIndicatorComponent,
} from '../../components';

const Home = () => {
  return (
    <ViewContainer>
      <LoadingIndicatorComponent />
      <SearchBarComponent />
      <FlatListComponent />
    </ViewContainer>
  );
};

export default Home;
