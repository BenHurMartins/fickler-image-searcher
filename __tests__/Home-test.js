import 'react-native';
import React from 'react';
import Home from '../src/scenes/Home';
import renderer from 'react-test-renderer';
describe('rendering test for Home.js', () => {
  it('Home renders correcly', () => {
    renderer.create(<Home />);
  });
  it('Home snapshot test', () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
