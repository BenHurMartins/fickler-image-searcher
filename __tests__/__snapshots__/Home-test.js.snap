// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering test for Home.js Home snapshot test 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "flex": 1,
        "width": 750,
      },
      undefined,
    ]
  }
>
  <Modal
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    theme={
      Object {
        "colors": Object {
          "black": "#242424",
          "disabled": "hsl(208, 8%, 90%)",
          "divider": "#bcbbc1",
          "error": "#ff190c",
          "grey0": "#393e42",
          "grey1": "#43484d",
          "grey2": "#5e6977",
          "grey3": "#86939e",
          "grey4": "#bdc6cf",
          "grey5": "#e1e8ee",
          "greyOutline": "#bbb",
          "platform": Object {
            "android": Object {
              "error": "#f44336",
              "grey": "rgba(0, 0, 0, 0.54)",
              "primary": "#2196f3",
              "searchBg": "#dcdce1",
              "secondary": "#9C27B0",
              "success": "#4caf50",
              "warning": "#ffeb3b",
            },
            "default": Object {
              "error": "#ff3b30",
              "grey": "#7d7d7d",
              "primary": "#007aff",
              "searchBg": "#dcdce1",
              "secondary": "#5856d6",
              "success": "#4cd964",
              "warning": "#ffcc00",
            },
            "ios": Object {
              "error": "#ff3b30",
              "grey": "#7d7d7d",
              "primary": "#007aff",
              "searchBg": "#dcdce1",
              "secondary": "#5856d6",
              "success": "#4cd964",
              "warning": "#ffcc00",
            },
            "web": Object {
              "error": "#ff190c",
              "grey": "#393e42",
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#ca71eb",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          },
          "primary": "#2089dc",
          "searchBg": "#303337",
          "secondary": "#ca71eb",
          "success": "#52c41a",
          "warning": "#faad14",
          "white": "#ffffff",
        },
      }
    }
    transparent={true}
    visible={false}
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(0, 0, 0, .4)",
          "height": "100%",
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "top": 0,
          "width": "100%",
        }
      }
      testID="RNE__Overlay__backdrop"
    />
    <View
      pointerEvents="box-none"
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderRadius": 3,
            "padding": 10,
            "shadowColor": "rgba(0, 0, 0, .3)",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowRadius": 4,
          }
        }
      >
        <ActivityIndicator
          animating={true}
          color="#000000"
          hidesWhenStopped={true}
          size="large"
        />
      </View>
    </View>
  </Modal>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "zIndex": 2,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "borderBottomColor": "transparent",
          "borderBottomWidth": 1,
          "borderTopColor": "transparent",
          "borderTopWidth": 1,
          "padding": 8,
          "width": "80%",
        }
      }
    >
      <View
        style={
          Object {
            "paddingHorizontal": 0,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderBottomColor": "#000000",
              "borderBottomWidth": 1,
              "borderColor": "#000000",
              "borderRadius": 3,
              "borderWidth": 1,
              "flexDirection": "row",
              "minHeight": 30,
              "overflow": "hidden",
              "transform": Array [
                Object {
                  "translateX": 0,
                },
              ],
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "height": 40,
                "justifyContent": "center",
                "marginLeft": 8,
                "marginVertical": 4,
                "paddingRight": 4,
              }
            }
          >
            <View
              style={
                Object {
                  "overflow": "hidden",
                }
              }
            >
              <View>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text
                    allowFontScaling={false}
                    style={
                      Array [
                        Object {
                          "color": "#86939e",
                          "fontSize": 18,
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                        Object {
                          "fontFamily": "Material Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                    testID="iconIcon"
                  >
                    
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <TextInput
            allowFontScaling={true}
            blurOnSubmit={true}
            editable={true}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            onSubmitEditing={[Function]}
            placeholder="Search for ..."
            placeholderTextColor="#86939e"
            platform="default"
            rejectResponderTermination={true}
            style={
              Object {
                "color": "#000000",
                "flex": 1,
                "fontSize": 18,
                "marginLeft": 10,
                "minHeight": 40,
              }
            }
            testID="input"
            underlineColorAndroid="transparent"
            value="test"
          />
          <View
            style={
              Object {
                "alignItems": "center",
                "height": 40,
                "justifyContent": "center",
                "marginRight": 8,
                "marginVertical": 4,
                "paddingRight": 4,
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <View
                style={
                  Object {
                    "overflow": "hidden",
                  }
                }
              >
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "transparent",
                        "justifyContent": "center",
                      }
                    }
                  >
                    <Text
                      allowFontScaling={false}
                      style={
                        Array [
                          Object {
                            "color": "#86939e",
                            "fontSize": 18,
                          },
                          Object {
                            "backgroundColor": "transparent",
                          },
                          Object {
                            "fontFamily": "Material Icons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                      testID="iconIcon"
                    >
                      
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
        <Text
          style={
            Object {
              "color": "#ff190c",
              "fontSize": 12,
              "height": 0,
              "margin": 0,
              "padding": 0,
            }
          }
        />
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#000000",
          "borderRadius": 5,
          "height": 50,
          "justifyContent": "center",
          "opacity": 1,
          "width": "15%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontSize": 30,
          }
        }
      >
        Go
      </Text>
    </View>
  </View>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "justifyContent": "space-around",
        "width": 750,
        "zIndex": 1,
      }
    }
    data={
      Array [
        Object {
          "farm": 1,
          "id": "1",
          "secret": "a",
          "server": "1",
        },
        Object {
          "farm": 2,
          "id": "2",
          "secret": "b",
          "server": "2",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onEndReachedThreshold={0.7}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Modal
              hardwareAccelerated={false}
              onRequestClose={[Function]}
              theme={
                Object {
                  "colors": Object {
                    "black": "#242424",
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "grey": "rgba(0, 0, 0, 0.54)",
                        "primary": "#2196f3",
                        "searchBg": "#dcdce1",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "default": Object {
                        "error": "#ff3b30",
                        "grey": "#7d7d7d",
                        "primary": "#007aff",
                        "searchBg": "#dcdce1",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "grey": "#7d7d7d",
                        "primary": "#007aff",
                        "searchBg": "#dcdce1",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                      "web": Object {
                        "error": "#ff190c",
                        "grey": "#393e42",
                        "primary": "#2089dc",
                        "searchBg": "#303337",
                        "secondary": "#ca71eb",
                        "success": "#52c41a",
                        "warning": "#faad14",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#ca71eb",
                    "success": "#52c41a",
                    "warning": "#faad14",
                    "white": "#ffffff",
                  },
                }
              }
              transparent={true}
              visible={false}
            >
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "rgba(0, 0, 0, .4)",
                    "height": "100%",
                    "justifyContent": "center",
                    "left": 0,
                    "position": "absolute",
                    "top": 0,
                    "width": "100%",
                  }
                }
                testID="RNE__Overlay__backdrop"
              />
              <View
                pointerEvents="box-none"
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                    "justifyContent": "center",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "white",
                      "borderRadius": 3,
                      "padding": 10,
                      "shadowColor": "rgba(0, 0, 0, .3)",
                      "shadowOffset": Object {
                        "height": 1,
                        "width": 0,
                      },
                      "shadowRadius": 4,
                    }
                  }
                >
                  <Image
                    height={730}
                    resizeMode="contain"
                    source={
                      Object {
                        "uri": "https://farm1.static.flickr.com/1/1_a.jpg",
                      }
                    }
                    style={
                      Object {
                        "height": 850,
                        "width": 730,
                      }
                    }
                    width={730}
                  />
                </View>
              </View>
            </Modal>
            <View
              style={
                Object {
                  "backgroundColor": "#ffffff",
                  "borderColor": "#000000",
                  "borderWidth": 1,
                  "margin": 10,
                  "marginBottom": 10,
                  "marginTop": 10,
                  "padding": 3,
                  "shadowColor": "rgba(0,0,0, .2)",
                  "shadowOffset": Object {
                    "height": 0,
                    "width": 0,
                  },
                  "shadowOpacity": 1,
                  "shadowRadius": 1,
                  "width": 350,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "transparent",
                  }
                }
              >
                <View
                  accessibilityIgnoresInvertColors={true}
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "overflow": "hidden",
                      "position": "relative",
                    }
                  }
                >
                  <Image
                    onLoad={[Function]}
                    resizeMode="contain"
                    source={
                      Object {
                        "uri": "https://farm1.static.flickr.com/1/1_a.jpg",
                      }
                    }
                    style={
                      Object {
                        "bottom": 0,
                        "height": 150,
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                        "width": null,
                      }
                    }
                    testID="RNE__Image"
                    theme={
                      Object {
                        "colors": Object {
                          "black": "#242424",
                          "disabled": "hsl(208, 8%, 90%)",
                          "divider": "#bcbbc1",
                          "error": "#ff190c",
                          "grey0": "#393e42",
                          "grey1": "#43484d",
                          "grey2": "#5e6977",
                          "grey3": "#86939e",
                          "grey4": "#bdc6cf",
                          "grey5": "#e1e8ee",
                          "greyOutline": "#bbb",
                          "platform": Object {
                            "android": Object {
                              "error": "#f44336",
                              "grey": "rgba(0, 0, 0, 0.54)",
                              "primary": "#2196f3",
                              "searchBg": "#dcdce1",
                              "secondary": "#9C27B0",
                              "success": "#4caf50",
                              "warning": "#ffeb3b",
                            },
                            "default": Object {
                              "error": "#ff3b30",
                              "grey": "#7d7d7d",
                              "primary": "#007aff",
                              "searchBg": "#dcdce1",
                              "secondary": "#5856d6",
                              "success": "#4cd964",
                              "warning": "#ffcc00",
                            },
                            "ios": Object {
                              "error": "#ff3b30",
                              "grey": "#7d7d7d",
                              "primary": "#007aff",
                              "searchBg": "#dcdce1",
                              "secondary": "#5856d6",
                              "success": "#4cd964",
                              "warning": "#ffcc00",
                            },
                            "web": Object {
                              "error": "#ff190c",
                              "grey": "#393e42",
                              "primary": "#2089dc",
                              "searchBg": "#303337",
                              "secondary": "#ca71eb",
                              "success": "#52c41a",
                              "warning": "#faad14",
                            },
                          },
                          "primary": "#2089dc",
                          "searchBg": "#303337",
                          "secondary": "#ca71eb",
                          "success": "#52c41a",
                          "warning": "#faad14",
                          "white": "#ffffff",
                        },
                      }
                    }
                    transition={true}
                    transitionDuration={360}
                  />
                  <View
                    accessibilityElementsHidden={true}
                    importantForAccessibility="no-hide-descendants"
                    pointerEvents="none"
                    style={
                      Object {
                        "bottom": 0,
                        "left": 0,
                        "opacity": 1,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#bdbdbd",
                          "height": 150,
                          "justifyContent": "center",
                          "width": null,
                        }
                      }
                      testID="RNE__Image__placeholder"
                    />
                  </View>
                  <View
                    style={
                      Object {
                        "height": 150,
                        "width": null,
                      }
                    }
                    testID="RNE__Image__children__container"
                  />
                </View>
              </View>
            </View>
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Modal
              hardwareAccelerated={false}
              onRequestClose={[Function]}
              theme={
                Object {
                  "colors": Object {
                    "black": "#242424",
                    "disabled": "hsl(208, 8%, 90%)",
                    "divider": "#bcbbc1",
                    "error": "#ff190c",
                    "grey0": "#393e42",
                    "grey1": "#43484d",
                    "grey2": "#5e6977",
                    "grey3": "#86939e",
                    "grey4": "#bdc6cf",
                    "grey5": "#e1e8ee",
                    "greyOutline": "#bbb",
                    "platform": Object {
                      "android": Object {
                        "error": "#f44336",
                        "grey": "rgba(0, 0, 0, 0.54)",
                        "primary": "#2196f3",
                        "searchBg": "#dcdce1",
                        "secondary": "#9C27B0",
                        "success": "#4caf50",
                        "warning": "#ffeb3b",
                      },
                      "default": Object {
                        "error": "#ff3b30",
                        "grey": "#7d7d7d",
                        "primary": "#007aff",
                        "searchBg": "#dcdce1",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                      "ios": Object {
                        "error": "#ff3b30",
                        "grey": "#7d7d7d",
                        "primary": "#007aff",
                        "searchBg": "#dcdce1",
                        "secondary": "#5856d6",
                        "success": "#4cd964",
                        "warning": "#ffcc00",
                      },
                      "web": Object {
                        "error": "#ff190c",
                        "grey": "#393e42",
                        "primary": "#2089dc",
                        "searchBg": "#303337",
                        "secondary": "#ca71eb",
                        "success": "#52c41a",
                        "warning": "#faad14",
                      },
                    },
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#ca71eb",
                    "success": "#52c41a",
                    "warning": "#faad14",
                    "white": "#ffffff",
                  },
                }
              }
              transparent={true}
              visible={false}
            >
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "rgba(0, 0, 0, .4)",
                    "height": "100%",
                    "justifyContent": "center",
                    "left": 0,
                    "position": "absolute",
                    "top": 0,
                    "width": "100%",
                  }
                }
                testID="RNE__Overlay__backdrop"
              />
              <View
                pointerEvents="box-none"
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                    "justifyContent": "center",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "white",
                      "borderRadius": 3,
                      "padding": 10,
                      "shadowColor": "rgba(0, 0, 0, .3)",
                      "shadowOffset": Object {
                        "height": 1,
                        "width": 0,
                      },
                      "shadowRadius": 4,
                    }
                  }
                >
                  <Image
                    height={730}
                    resizeMode="contain"
                    source={
                      Object {
                        "uri": "https://farm2.static.flickr.com/2/2_b.jpg",
                      }
                    }
                    style={
                      Object {
                        "height": 850,
                        "width": 730,
                      }
                    }
                    width={730}
                  />
                </View>
              </View>
            </Modal>
            <View
              style={
                Object {
                  "backgroundColor": "#ffffff",
                  "borderColor": "#000000",
                  "borderWidth": 1,
                  "margin": 10,
                  "marginBottom": 10,
                  "marginTop": 10,
                  "padding": 3,
                  "shadowColor": "rgba(0,0,0, .2)",
                  "shadowOffset": Object {
                    "height": 0,
                    "width": 0,
                  },
                  "shadowOpacity": 1,
                  "shadowRadius": 1,
                  "width": 350,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "transparent",
                  }
                }
              >
                <View
                  accessibilityIgnoresInvertColors={true}
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "overflow": "hidden",
                      "position": "relative",
                    }
                  }
                >
                  <Image
                    onLoad={[Function]}
                    resizeMode="contain"
                    source={
                      Object {
                        "uri": "https://farm2.static.flickr.com/2/2_b.jpg",
                      }
                    }
                    style={
                      Object {
                        "bottom": 0,
                        "height": 150,
                        "left": 0,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                        "width": null,
                      }
                    }
                    testID="RNE__Image"
                    theme={
                      Object {
                        "colors": Object {
                          "black": "#242424",
                          "disabled": "hsl(208, 8%, 90%)",
                          "divider": "#bcbbc1",
                          "error": "#ff190c",
                          "grey0": "#393e42",
                          "grey1": "#43484d",
                          "grey2": "#5e6977",
                          "grey3": "#86939e",
                          "grey4": "#bdc6cf",
                          "grey5": "#e1e8ee",
                          "greyOutline": "#bbb",
                          "platform": Object {
                            "android": Object {
                              "error": "#f44336",
                              "grey": "rgba(0, 0, 0, 0.54)",
                              "primary": "#2196f3",
                              "searchBg": "#dcdce1",
                              "secondary": "#9C27B0",
                              "success": "#4caf50",
                              "warning": "#ffeb3b",
                            },
                            "default": Object {
                              "error": "#ff3b30",
                              "grey": "#7d7d7d",
                              "primary": "#007aff",
                              "searchBg": "#dcdce1",
                              "secondary": "#5856d6",
                              "success": "#4cd964",
                              "warning": "#ffcc00",
                            },
                            "ios": Object {
                              "error": "#ff3b30",
                              "grey": "#7d7d7d",
                              "primary": "#007aff",
                              "searchBg": "#dcdce1",
                              "secondary": "#5856d6",
                              "success": "#4cd964",
                              "warning": "#ffcc00",
                            },
                            "web": Object {
                              "error": "#ff190c",
                              "grey": "#393e42",
                              "primary": "#2089dc",
                              "searchBg": "#303337",
                              "secondary": "#ca71eb",
                              "success": "#52c41a",
                              "warning": "#faad14",
                            },
                          },
                          "primary": "#2089dc",
                          "searchBg": "#303337",
                          "secondary": "#ca71eb",
                          "success": "#52c41a",
                          "warning": "#faad14",
                          "white": "#ffffff",
                        },
                      }
                    }
                    transition={true}
                    transitionDuration={360}
                  />
                  <View
                    accessibilityElementsHidden={true}
                    importantForAccessibility="no-hide-descendants"
                    pointerEvents="none"
                    style={
                      Object {
                        "bottom": 0,
                        "left": 0,
                        "opacity": 1,
                        "position": "absolute",
                        "right": 0,
                        "top": 0,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#bdbdbd",
                          "height": 150,
                          "justifyContent": "center",
                          "width": null,
                        }
                      }
                      testID="RNE__Image__placeholder"
                    />
                  </View>
                  <View
                    style={
                      Object {
                        "height": 150,
                        "width": null,
                      }
                    }
                    testID="RNE__Image__children__container"
                  />
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
